packages/browser-wallet/src/popup/pages/Web3ProofRequest/Web3ProofRequest.tsx

```diff
@@ -0,0 +1,186 @@
+import { displayUrl } from '@popup/shared/utils/string-helpers';
+import {
+    storedVerifiableCredentialsAtom,
+    storedVerifiableCredentialSchemasAtom,
+} from '@popup/store/verifiable-credential';
+import { useConfirmedIdentities } from '@popup/shared/utils/identity-helpers';
+import { parse } from '@shared/utils/payload-helpers';
+import { DisplayCredentialStatement } from './DisplayStatement';
+import { getCommitmentInput } from './utils';
+
+type Props = {
+    onSubmit(presentationString: string): void;
+    onReject(): void;
+};
+
+interface Location {
+    state: {
+        payload: {
+            challenge: string;
+            statements: string;
+            url: string;
+        };
+    };
+}
+
+export default function Web3ProofRequest({ onReject, onSubmit }: Props) {
+    const { state } = useLocation() as Location;
+    const { statements: rawStatements, challenge, url } = state.payload;
+    const { onClose, withClose } = useContext(fullscreenPromptContext);
+    const { t } = useTranslation('web3IdProofRequest');
+    const network = useAtomValue(networkConfigurationAtom);
+    const client = useAtomValue(grpcClientAtom);
+    const addToast = useSetAtom(addToastAtom);
+    const recoveryPhrase = useDecryptedSeedPhrase((e) => addToast(e.message));
+    const dappName = displayUrl(url);
+    const [creatingProof, setCreatingProof] = useState<boolean>(false);
+    const [currentStatementIndex, setCurrentStatementIndex] = useState<number>(0);
+    const net = getNet(network);
+
+    const statements: CredentialStatements = useMemo(() => parse(rawStatements), [rawStatements]);
+
+    const [ids, setIds] = useState<string[]>(Array(statements.length).fill(''));
+
+    const verifiableCredentialSchemas = useAtomValue(storedVerifiableCredentialSchemasAtom);
+    const identities = useConfirmedIdentities();
+    const credentials = useAtomValue(credentialsAtom);
+    const verifiableCredentials = useAtomValue(storedVerifiableCredentialsAtom);
+
+    const canProve = useMemo(() => ids.every((x) => Boolean(x)), [ids]);
+
+    const handleSubmit = useCallback(async () => {
+        if (!recoveryPhrase) {
+            throw new Error('Missing recovery phrase');
+        }
+        if (!network) {
+            throw new Error('Network is not specified');
+        }
+        if (!ids.every((x) => Boolean(x))) {
+            throw new Error('Network is not specified');
+        }
+
+        const global = await getGlobal(client);
+        const wallet = ConcordiumHdWallet.fromHex(recoveryPhrase, net);
+
+        const type = ['ConcordiumVerifiableCredential', 'TestCredential', 'VerifiableCredential'];
+
+        const parsedStatements: RequestStatement[] = statements.map((statement, index) => {
+            if (isAccountCredentialStatement(statement)) {
+                return { statement: statement.statement, id: ids[index] };
+            }
+            return { statement: statement.statement, id: ids[index], type };
```

Comment:
I made a comment somewhere else where we differentiate the types of `RequestStatement` based on this `type` field. I think it would be nicer to make it more explicit with `enum` value or something of that sort.

Reply:
I will add a helper to the SDK.